src/unixdomainsocket.rs

```diff
@@ -82,8 +82,9 @@ pub struct RxToken {
 }
 
 impl phy::RxToken for RxToken {
-    fn consume<R, F: FnOnce(&[u8]) -> Result<R>>(self, _timestamp: Instant, f: F) -> Result<R> {
-        f(&self.buffer[..])
+    fn consume<R, F: FnOnce(&mut [u8]) -> Result<R>>(self, _timestamp: Instant, f: F) -> Result<R> {
+        let mut buffer = self.buffer.clone();
+        f(&mut buffer[..])
```

Comment:
Thanks for the PR! I just noticed it by chance, sorry for the delay.
Do you think we could spare the `clone` here somehow?

Reply:
Found it: instead of introducing the clone, directly use `&mut self.buffer` but also mark it above as mutable: `mut self, _timestamp…`